feat: add rule prefer-inline-action-props

rafaelss95 opened the pull request:
Closes #66.

rafaelss95 commented on src/rules/action-prop-name.ts:
```diff
@@ -0,0 +1,50 @@
+
+type Options = []
+
+export default ESLintUtils.RuleCreator(docsUrl)<Options, MessageIds>({
```
Let me know what do you think about the rule name and its description/failures.

timdeschryver replied:
What do you think about prefer-inline-action-props ?

rafaelss95 replied:
Done. What about the description and failure messages?

timdeschryver replied:
I think it's good.  👍
We'll provide more info in the docs folder soonish

timdeschryver commented:
@rafaelss95 thanks for the contribution 🙂
I just wanted to check if you agree with the rename to `prefer-inline-action-props`. If so, do you have to make the changes? Otherwise, I could also do them.

rafaelss95 commented:
Hey @timdeschryver don't worry, I'll do it asap.

rafaelss95 commented on tests/rules/action-prop-name.test.ts:
```diff
@@ -0,0 +1,91 @@
+    `const ok0 = createAction('ok0', props<{ id: number, name: string }>())`,
+    `const ok1 = createAction('ok1', props<Readonly<{ description: string }>>())`,
+    `const ok2 = createAction('ok2', props<Readonly<HttpErrorResponse & { description: string }>>())`,
+    `const saveUserSuccess = createAction('[API/User] Save user success', (user: User<number>) => ({
```
Good catch, I'll add it soon.

timdeschryver commented:
Take your time, there's no rush @rafaelss95 🙂
I didn't mean to rush you.

rafaelss95 commented:
> Take your time, there's no rush @rafaelss95 🙂
> I didn't mean to rush you.

Ah, nvm, that's fine 😄 
I didn't do it yesterday because I was with a lot of work.

I renamed it to `prefer-inline-action-props` and also added the missing test case you mentioned. Please take a look and let me know if it's everything fine now.
